src/clients/entities/client.entity.ts

```diff
@@ -0,0 +1,44 @@
+import { ObjectType, Field, Int } from '@nestjs/graphql';
+import { IsBoolean, IsEmail, IsPhoneNumber, IsString } from 'class-validator';
+import mongoose from 'mongoose';
+
+export type ClientDocument = Client & Document;
+export const ClientSchema = new mongoose.Schema(
+  {
```

Comment:
Please change code : 
import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
import { HydratedDocument } from 'mongoose';

export type CatDocument = HydratedDocument<Cat>; // you can remove this line

@Schema()
export class Cat { // CatDocument
  @Prop()
  name: string;

  @Prop()
  age: number;

  @Prop()
  breed: string;
}

export const CatSchema = SchemaFactory.createForClass(Cat); // CatDocument